python/lsst/ip/diffim/utils.py

```diff
@@ -711,3 +732,354 @@ def plotWhisker(results, newWcs):
+
+        """
+
+        if diffim is None:
+            diffim = self.diffim
+
+        # Start with a minimal schema - only the fields all SourceCatalogs need
+        schema = afwTable.SourceTable.makeMinimalSchema()
+
+        # Customize the detection task a bit (optional)
+        detectConfig = measAlg.SourceDetectionConfig()
+        detectConfig.returnOriginalFootprints = False  # should be the default
+
+        psfSigma = diffim.getPsf().computeShape().getDeterminantRadius()
+
+        # code from imageDifference.py:
+        detectConfig.thresholdPolarity = "both"
+        detectConfig.thresholdValue = detectSigma
+        # detectConfig.nSigmaToGrow = psfSigma
+        detectConfig.reEstimateBackground = True  # if False, will fail often for faint sources on gradients?
+        detectConfig.thresholdType = "pixel_stdev"
+
+        # Create the detection task. We pass the schema so the task can declare a few flag fields
+        detectTask = measAlg.SourceDetectionTask(schema, config=detectConfig)
+
+        table = afwTable.SourceTable.make(schema)
+        catalog = detectTask.makeSourceCatalog(table, diffim, sigma=psfSigma)
+
+        # Now do the merge.
+        if doMerge:
+            fpSet = catalog.fpSets.positive
+            fpSet.merge(catalog.fpSets.negative, grow, grow, False)
+            sources = afwTable.SourceCatalog(table)
+            fpSet.makeSources(sources)
+
+            return sources
+
+        else:
+            return detectTask, schema
+
+
+# ######## UTILITIES FUNCTIONS FOR PLOTTING  ####
+
+# Utility class containing methods for displaying dipoles/footprints
+# in difference images, mostly used for debugging.
+
+
+def display2dArray(arr, title='Data', showBars=True, extent=None):
+    """Use matplotlib.pyplot.imshow() to display a 2-D array.
+
+    @param arr The 2-D array to display
+    @param title Optional title to display
+    @param showBars Show grey-scale bar alongside
+    @param extent If not None, a 4-tuple giving the bounding box coordinates of the array
+
+    @return matplotlib.pyplot.figure dispaying the image
+    """
+    plt = importMatplotlib()
+    if not plt:
+        return
+
+    fig = plt.imshow(arr, origin='lower', interpolation='none', cmap='gray',
+                     extent=extent)
+    plt.title(title)
+    if showBars:
+        plt.colorbar(fig, cmap='gray')
+    return fig
+
+
+def displayImage(image, showBars=True, width=8, height=2.5):
+    """Use matplotlib.pyplot.imshow() to display an afw.image.Image within
+    its bounding box.
+
+    @see displayImages()
+    """
+    return displayImages((image), showBars=True, width=8, height=2.5)
+
+
+def displayImages(images, showBars=True, width=8, height=2.5):
+    """Use matplotlib.pyplot.imshow() to display up to three
+    afw.image.Images alongside each other, each within its
+    bounding box.
+
+    @see display2dArray() for params not listed here.
+    @param images tuple of up to three images to display
```

Comment:
The idea here is to display up to 3 images in one "row" of a plot (side-by-side), without having to call the `matplotlib` magic to set up the subplots and increment them for each image. But you are right, I will restructure this code and make it explicit that it can take up to 3 images and no more.